proxy/src/avatar.rs

```diff
@@ -0,0 +1,201 @@
+//! Org and user avatar generation.
+
+/// Emoji whitelist.
+const EMOJIS: &[&str] = &[
+    "⌚️", "📱", "📲", "💻", "⌨️", "💽", "💾", "💿", "📀", "📼", "📷", "📸", "📹", "🎥", "🎞", "📞",
+    "☎️", "📟", "📠", "📺", "📻", "⏰", "🕰", "⌛️", "⏳", "📡", "🔋", "🔌", "💡", "🔦", "💸", "💵",
+    "💴", "💶", "💷", "💰", "💳", "🧾", "💎", "⚖️", "🔧", "🔨", "", "🔩", "⚙️", "⛓", "🔫", "💣", "🔪",
+    "🗡", "⚔️", "🚬", "⚰️", "⚱️", "🏺", "🧭", "🧱", "🔮", "🧿", "🧸", "📿", "💈", "⚗️", "🔭", "🧰", "🧲", "🧪",
+    "🧫", "🧬", "🧯", "🔬", "🕳", "💊", "💉", "🚽", "🚰", "🚿", "🛁", "🛀", "🛀🏻", "🛀🏼", "🛀🏽",
+    "🛀🏾", "🛀🏿", "🧴", "🧵", "🧶", "🧷", "🧹", "🧺", "🧻", "🧼", "🧽", "🛎", "🔑", "🗝", "🚪", "🛍", "🧳",
+    "🛒", "🎁", "🎈", "🎏", "🎀", "🎊", "🎉", "🧨", "🎎", "🏮", "🎐", "🧧", "✉️", "📩", "📨", "📧",
+    "💌", "📥", "📤", "📦", "📪", "📫", "📬", "📭", "📮", "📯", "📜", "📃", "📄", "📑", "📊", "📈",
+    "📉", "📆", "📅", "📇", "📋", "📁", "📂", "🗂", "🗞", "📰", "📓", "📔", "📒", "📕", "📗", "📘",
+    "📙", "📚", "📖", "🔖", "🔗", "📎", "📐", "📏", "📌", "📍", "✂️", "🖊", "🖋", "✒️", "🖌", "📝", "✏️",
+    "🔍", "🔎", "🔏", "🔐",
+];
+
+/// An emoji.
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub struct Emoji(&'static str);
+
+/// An avatar.
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub struct Avatar {
+    /// The emoji component.
+    pub emoji: Emoji,
+    /// The background color component.
+    pub background: Color,
+}
+
+impl Avatar {
+    /// Generate an avatar from an input string.
+    pub fn from(input: &str) -> Self {
+        Self {
+            emoji: generate_emoji(input),
+            background: compress_color(generate_color(input)),
+        }
+    }
+}
+
+/// A 32-bit RGBA color.
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub struct Color {
```

Comment:
```suggestion
pub struct RGBA {
```

Could communicate the type better, but both work.

Reply:
I considered it, I really did. But I like `Color` because it communicates the purpose a little better.

Reply:
:smirk_cat: